File: trunk/java/JavaMercFramework/src/main/it/mintlab/desktopjava/mercframework/MessageContent.java
package it.mintlab.desktopjava.mercframework;

import it.mintlab.desktopjava.tupleengine.Term;
import it.mintlab.desktopjava.tupleengine.TupleEngine;

	public class MessageContent
	{
		private Category category = Category.forValue(0);
		private Term tuple;

		public enum Category
		{
			COMMAND,
			ERROR,
			LOG;

			public int getValue()
			{
				return this.ordinal();
			}

			public static Category forValue(int value)
			{
				return values()[value];
			}
		}

		public MessageContent(Category category, String tuple)
		{
			this(category, TupleEngine.parse(tuple));
		}

		public MessageContent(Category category, String tuple, Object[] pars)
		{
			this(category, TupleEngine.parse(tuple, pars));
		}

		public MessageContent(Category category, Term tuple)
		{
			this.category = category;
			this.tuple = tuple;
		}

		public final Category getCategory()
		{
			return category;
		}

		public final Term getTuple()
		{
			return tuple;
		}

		@Override
		public boolean equals(Object obj)
		{
			MessageContent mc = (MessageContent)obj;
			return (category == mc.getCategory() && tuple.match(mc.getTuple()));
		}
		
		@Override
		public int hashCode()
		{
			return getKey().hashCode();
		}

		@Override
		public String toString()
		{
			return tuple + " [" + category + "]";
		}

		public final String getKey()
		{
			String functor = tuple.toString().split(java.util.regex.Pattern.quote("("))[0]; //return the first part of tuple
			return "" + category + functor;
		}
	}

File: compact-net/CompactNetMercFramework/MessageContent.cs
using System;
using System.Collections.Generic;
using System.Text;
using it.mintlab.mobilenet.tupleengine;

namespace it.mintlab.mobilenet.mercframework
{
    public class MessageContent
    {
        private Category category;
        private Term tuple;

        public enum Category
        {
            COMMAND,
            ERROR,
            LOG
        };


        public MessageContent(Category category, string tuple)
            : this(category, TupleEngine.parse(tuple))
        { }

        public MessageContent(Category category, string tuple, object[] pars)
            : this(category, TupleEngine.parse(tuple, pars))
        { }

        public MessageContent(Category category, Term tuple)
        {
            this.category = category;
            this.tuple = tuple;
        }

        public Category getCategory()
        {
            return category;
        }

        public Term getTuple()
        {
            return tuple;
        }

        public override bool Equals(object obj)
        {
            MessageContent mc = (MessageContent)obj;
            return category == mc.getCategory() && tuple.match(mc.getTuple());
        }

        public override int GetHashCode()
        {
            return getKey().GetHashCode();
        }

        public override string ToString()
        {
            return tuple + " [" + category + "]";
        }

        internal string getKey()
        {
            string functor = tuple.ToString().Split(new char[] { '(' })[0]; //return the first part of tuple
            return "" + category + functor;
        }

    }
}
